async_tcp_server.cpp

```diff
@@ -0,0 +1,101 @@
+#include "async_tcp_server.hpp"
+#include <algorithm>
+#include <cstring>
+#include <iostream>
+#include <sys/select.h>
+
+AsyncTCPServer::AsyncTCPServer() {}
+
+AsyncTCPServer::AsyncTCPServer(Socket_sptr master_socket, fd_set fdsetRead): m_fdsetRead(fdsetRead)
+{
+    FD_ZERO(&m_fdsetRead);
+    FD_SET(master_socket->GetSocketHandle(), &m_fdsetRead);
+    m_aSockets.push_back(master_socket);
+}
+
+Socket_sptr AsyncTCPServer::GetSocketByFD(int fd) const
+{
+    for(auto socket : m_aSockets)
+    {
+        if (socket->GetSocketHandle() == fd)
```

Comment:
Also it will be really useful to have comparison operators in `Socket` class. Maybe even two overloads: `Socket == Socket` and `Socket == FD` (`int` in our case)